### PR TITLE
Replace all internal `ZIO.fail(None)` with `Exit.failNone`

### PR DESCRIPTION
**Pull request by guizmaii**

### CODE REVIEW

File: streams/jvm/src/main/scala/zio/stream/platform.scala [hunk 1]
```diff
@@ -237,7 +237,7 @@ private[stream] trait ZStreamPlatformSpecificConstructors {
             ZStream.repeatZIOChunkOption(
               for {
                 bytesRead <- ZIO.attempt(channel.read(reusableBuffer)).asSomeError
-                _         <- ZIO.fail(None).when(bytesRead == -1)
+                _         <- Exit.failNone.when(bytesRead == -1)
```
Comment (kyri-petrou) [inline streams/jvm/src/main/scala/zio/stream/platform.scala]: A bit outside of the scope of this PR, but since we're at it might change these usages to `whenDiscard`

File: streams/shared/src/main/scala/zio/stream/ZStream.scala [hunk 2]
```diff
@@ -3491,8 +3491,8 @@ final class ZStream[-R, +E, +A] private (val channel: ZChannel[R, Any, Any, Any,
   def toPull(implicit trace: Trace): ZIO[R with Scope, Nothing, ZIO[R, Option[E], Chunk[A]]] =
     channel.toPull.map { pull =>
       pull.mapError(error => Some(error)).flatMap {
-        case Left(done)  => ZIO.fail(None)
-        case Right(elem) => ZIO.succeed(elem)
+        case Left(done)  => Exit.failNone
```
Comment (kyri-petrou) [inline streams/shared/src/main/scala/zio/stream/ZStream.scala]: Let's use `foldZIO` instead of `mapError` and `flatMap` in the line above this while we're at it

File: streams/shared/src/main/scala/zio/stream/ZStream.scala [hunk 3]
```diff
@@ -4318,7 +4318,7 @@ object ZStream extends ZStreamPlatformSpecificConstructors {
           bufArray  <- ZIO.succeed(Array.ofDim[Byte](chunkSize))
           bytesRead <- ZIO.attemptBlockingIO(is.read(bufArray)).asSomeError
           bytes <- if (bytesRead < 0)
-                     ZIO.fail(None)
+                     Exit.failNone
```
Comment (kyri-petrou) [inline streams/shared/src/main/scala/zio/stream/ZStream.scala:4324]: Maybe better done as a separate PR, but maybe we should store `Exit.succeed(Chunk.empty)` in a val and use it below here

**Review by kyri-petrou — COMMENTED** [review]
I know the PR is still in draft but love it already! Just some minor comments most which we might want to address in followup PRs

File: streams/shared/src/main/scala/zio/stream/ZStream.scala (hunk 3 above)
Comment (guizmaii) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala:4324]: Hehe: https://github.com/zio/zio/pull/9360

File: streams/jvm/src/main/scala/zio/stream/platform.scala (hunk 1 above)
Comment (guizmaii) [reply streams/jvm/src/main/scala/zio/stream/platform.scala]: Done

File: streams/shared/src/main/scala/zio/stream/ZStream.scala (hunk 2 above)
Comment (guizmaii) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala]: Done

File: streams/shared/src/main/scala/zio/stream/ZStream.scala (hunk 3 above)
Comment (kyri-petrou) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala:4324]: Wow 😆

**Review by kyri-petrou — APPROVED** [review]
Thank you 🙏
